### ISSUE TITLE
SSL revocation error on Windows with corporate MITM certificate

### ISSUE THREAD

**Issue by Syrus-at-Philips** [mitsuhiko/rye#536]
### Steps to Reproduce

1. Download and install rye on a corporate machine running the Cisco Umbrella MITM proxy
2. Install in PowerShell under windows.
3. Attempt to do any install or update command that requires network access.

### Expected Result

The command will complete successfully.

### Actual Result

```
❯ rye self update
Updating to latest
error: could not download release latest for this platform

Caused by:
    0: download of https://github.com/mitsuhiko/rye/releases/latest/download/rye-x86_64-windows.exe failed
    1: [35] SSL connect error (schannel: next InitializeSecurityContext failed: Unknown error (0x80092012) - The revocation function was unable to check revocation for the certificate.)
```

### Version Info

```
rye 0.17.0
commit: 0.17.0 (500d8c4e6 2024-01-15)
platform: windows (x86_64)
self-python: cpython@3.11
symlink support: true
```

**Comment by mitsuhiko** [comment]
I feel like I need to understand a bit more. That flag is not exposed and really only implemented for windows on curl. Is the certificate of the MITM proxy not installed on your windows machine? Why is `--no-revoke-ssl` necessary in curl?

**Comment by Syrus-at-Philips** [comment]
The certificate for the MITM proxy is installed, and it's what signs the response after unwrapping and scanning. The accepted answer here is one source of the proposed workaround: https://stackoverflow.com/questions/54938026/curl-unknown-error-0x80092012-the-revocation-function-was-unable-to-check-r?r=Saves_UserSavesList

![image](https://github.com/mitsuhiko/rye/assets/119881585/14c4f4b3-a877-40aa-a158-a9ab92207ad0)

I used the no-revoke-ssl option in the conda configuration for mamba-forge, and it worked. In the case of the Cisco Umbrella, this does not impact the security of SSL connections because all connections are made by the proxy so it can scan them and then rewrap them in the local MITM cert. I'm no expert on this topic, but this is a workaround I found that applies to curl on Windows with this type of corporate MITM proxy.
